docs/docs/adrs/adr-017-allowing-inactive-validators.md

```diff
@@ -62,12 +62,16 @@ The following changes to the state are required:
 * Store the provider consensus validator set in the provider module state under the `LastProviderConsensusValsPrefix` key. This is the last set of validators that the provider sent to the consensus engine. This is needed to compute the ValUpdates to send to the consensus engine (by diffing the current set with this last sent set).
 * Increase the `MaxValidators` parameter of the staking module to the desired size of the potential validator
 set of consumer chains.
+* Introduce two extra per-consumer-chain parameters: `MinStake` and `MaxValidatorRank`. `MinStake` is the minimum amount of stake a validator must have to be considered for validation on the consumer chain. `MaxValidatorRank` is the maximum rank of a validator that can validate on the consumer chain. The provider module will only consider the first `MaxValidatorRank` validators that have at least `MinStake` stake as potential validators for the consumer chain.
 
 ## Risk Mitigations
 
 To mitigate risks from validators with little stake, we introduce a minimum stake requirement for validators to be able to validate on consumer chains, which can be set by each consumer chain independently, with a default value set by the provider chain.
 
-Additionally, we indepdently allow individual consumer chains to disable this feature, which will disallow validators from outside the provider active set from validating on the consumer chain and revert them to the previous behaviour of only considering validators of the provider that are part of the active consensus validator set.
+Additionally, we independently allow individual consumer chains to set a maximum rank for validators.
+This means that validators above a certain position in the validator set cannot validate on the consumer chain.
+Setting this to be equal to `MaxProviderConsensusValidators` effectively disables this feature,
```

Comment:
```suggestion
Setting `MaxValidatorRank` to be equal to `MaxProviderConsensusValidators` effectively disables this feature,
```

Can `MaxValidatorRank` be smaller than `MaxProviderConsensusValidators`?

Reply:
Yeah, don't see a reason why not, it just gives more expressive power to consumer chains. Maybe they think that the last 100 validators in the active set shouldn't be able to validate their chain; this allows them to express that.
They can for sure set parameters that will not make much sense, but I don't see a reason to forbid them programatically.

Reply:
Oh, I reread the text and realize why this might be confusing.